Source/player.cpp

```diff
@@ -515,26 +515,44 @@ void StartSpell(int pnum, Direction d, int cx, int cy)
 		return;
 	}
 
+	// Checks conditions for spell again, cause initial check was done when spell was queued and the parameters could be changed meanwhile
+	bool isValid = true;
+	switch (player.queuedSpell.spellType) {
+	case RSPLTYPE_SKILL:
+	case RSPLTYPE_SPELL:
+		isValid = CheckSpell(pnum, player.queuedSpell.spellId, player.queuedSpell.spellType, true) == SpellCheckResult::Success;
+		break;
+	case RSPLTYPE_SCROLL:
+		isValid = UseScroll(player, player.queuedSpell.spellId);
+		break;
+	case RSPLTYPE_CHARGES:
+		isValid = UseStaff(player, player.queuedSpell.spellId);
+		break;
+	}
+	if (!isValid)
+		return;

```

Comment:
It seems that runes are completely broken in this PR. I can't even put one down.

Reply:
Thank you very much for catching this one. 👍 🙂 
I should really test more in hellfire. 😢
Hopefully with the last change this should be fixed.

Reply:
I also don't usually test in Hellfire, and it's bitten me a few times.

It does appear that the issue is fixed, although this has the same issue as scrolls. In this case, however, Player 1 was able to kill Player 2 with a Greater Rune of Fire that Player 2 didn't even know had been placed, let alone triggered. 😓

Reply:
Yeah same issues and same benefits (double "casting"/placing a rune is fixed). 🙂